gdraw/gtabset.c

```diff
@@ -998,14 +1056,32 @@ void GTabSetRemoveTabByPos(GGadget *g, int pos) {
 	for ( i=pos+1; i<gts->tabcnt; ++i )
 	    gts->tabs[i-1] = gts->tabs[i];
 	--gts->tabcnt;
+	/* CharView keeps its own tabs array with its own private data.
+	 * This gives us a way to realign that array upon changes to ours. */
+	if (gts->remove_sync != NULL) (gts->remove_sync)(gts->g.base, pos);
 	if ( gts->sel==pos ) {
 	    if ( gts->sel==gts->tabcnt )
 		--gts->sel;
 	    GTabSetChanged(gts,pos);
 	}
     }
 }
-	
+
```

Comment:
This is definitely *easier*; it's a bit unfortunate that it's also *counter-intuitive*. Almost all draggable tabs I see these days allow the tab to be put between existing tabs. 

The data-structure implementation of "move between" isn't much more difficult -- one could just memcpy the values around in the array. So I take it the problem is/extra work would be providing some sort of visual feedback, like a color change to the lines separating the tabs as one drags into the "between region". 

In any case the project members may want to consider whether swapping is the right semantic.

Reply:
I already spent way too much time on this. I'm not willing to re-implement this, sorry. Swapping makes sense to me given the constraints that this is an entirely custom UI system.

Reply:
I don't disagree, but I'm also not going to unilaterally approve the PR for this reason, even if we resolve the (small number of) code-level questions. Other folks should have a chance to weigh in.

Reply:
I wouldn't ask you to give unilateral approval. Actually, I talked to @frank-trampe on a Skype call about the UI and implementation of both this and the COLR/CPAL patch (and my plans for that) and I am quite sure once we're done talking about the code itself he'll approve it since to my memory he had no problems with my plan for this PR.

Reply:
I felt it was worth pointing out but it's not like doing this now precludes doing something else later. Someone used to a swap semantic would maybe do a double-take once and then just use it the other way.